src/emucore/Random.cxx

```diff
@@ -24,26 +24,32 @@ void Random::seed(uInt32 value)
 {
   ourSeed = value;
   ourSeeded = true;
+  rndGenerator.seed(ourSeed);
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 Random::Random()
 {
+  maxPossibleValue = rndGenerator.max();
   // If we haven't been seeded then seed ourself
   if(!ourSeeded)
   {
-    cerr << "random seed " << endl;
-    ourSeed = (uInt32)time(0);
+    ourSeed = (uInt32)time(NULL);
     ourSeeded = true;
+    rndGenerator.seed(ourSeed);
   }
-
-  myValue = ourSeed;
 }
  
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 uInt32 Random::next()
 {
-  return (myValue = (myValue * 2416 + 374441) % 1771875);
+  return rndGenerator();
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+float Random::nextDouble()
+{
+  return rndGenerator() / double(maxPossibleValue + 1.0);
```

Comment:
I kept it consistent with the previous implementation, where it was being
used RAND_MAX + 1.

On 25 May 2015 at 12:03, Matthew Hausknecht <notifications@github.com>
wrote:

> In src/emucore/Random.cxx
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55#discussion_r30989746>
> :
>
> >  }
> >
> >  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
> >  uInt32 Random::next()
> >  {
> > -  return (myValue = (myValue * 2416 + 374441) % 1771875);
> > +  return rndGenerator();
> > +}
> > +
> > +// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
> > +float Random::nextDouble()
> > +{
> > +  return rndGenerator() / double(maxPossibleValue + 1.0);
>
> Why the + 1.0 here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55/files#r30989746>
> .
>



-- 
Marlos Cholodovskis Machado
http://www.dcc.ufmg.br/~marlos <http://www.cs.ualberta.ca/~machado>